### ISSUE TITLE
rustdoc: some fatal lexing errors are not buffered

### ISSUE THREAD

**Issue by euclio** [rust-lang/rust#56885]
Discovered while working on #56884.

For code like the following:

```
/// ```
/// "unterminated
/// ```
pub struct Foo;
```

rustdoc will fail to output any documentation for the struct and outputs a fatal error. I would expect the lexing error to be caught and emitted without stopping rendering.

**Comment by GuillaumeGomez** [comment]
Closing this issue since it has been fixed by #56884.

**Comment by euclio** [comment]
This wasn't fixed by that #56884. Current output:

```
$ rustdoc +nightly test.rs
error: unterminated double quote string
 --> <doctest>:1:1
  |
1 | "unterminated
  | ^^^^^^^^^^^^^^
```

This is a problem with the lexer.

**Comment by GuillaumeGomez** [comment]
Ah my bad. Reopening.
